internal/cli/board/list.go

```diff
@@ -47,6 +49,7 @@ func initListCommand() *cobra.Command {
 	}
 
 	timeoutArg.AddToCommand(listCommand)
+	fqbn.AddToCommandWithoutBoardOptions(listCommand)
```

Comment:
```suggestion
	fqbn.AddToCommand(listCommand)
```

The `--board-options` flag is actually applicable to this command. The reason is it is possible for boards definitions to associate custom board options with identification properties provided by a pluggable discovery tool. In this case `arduino-cli board list` will include the relevant board options in the FQBN:

https://arduino.github.io/arduino-cli/dev/pluggable-discovery-specification/#identification-of-board-options

Here is a demo of how Arduino CLI works with the suggested change:

```text
$ export ARDUINO_DIRECTORIES_DATA="/tmp/foo-data" # Use a throwaway directories.data for the demo.

$ arduino-cli core install arduino:avr@1.8.6

[...]

$ echo "mega.menu.cpu.atmega2560.upload_port.vid=0x2341" >> "${ARDUINO_DIRECTORIES_DATA}/packages/arduino/hardware/avr/1.8.6/boards.txt" # Simulate identification property

$ ./arduino-cli board list --fqbn arduino:avr:mega --board-options "cpu=atmega2560"
Port Protocol Type              Board Name                FQBN                            Core       
COM3 serial   Serial Port (USB) Arduino Mega or Mega 2560 arduino:avr:mega:cpu=atmega2560 arduino:avr
```

Reply:
Surely board options could be added here, but they bring a design questions: if the options partially match, is that an hit or a miss?

Reply:
What do we do in other cases?

Reply:
That decision would need to be made regardless because custom board options can be specified as part of the FQBN. Here is the behavior of `arduino-cli board list` from the PR in its current state (without my suggested modification):

```text
$ arduino-cli board list
Port Protocol Type              Board Name                FQBN                            Core       
COM1 serial   Serial Port       Unknown
COM3 serial   Serial Port (USB) Arduino Mega or Mega 2560 arduino:avr:mega:cpu=atmega2560 arduino:avr

$ arduino-cli board list --fqbn arduino:avr:mega
No boards found.

$ arduino-cli board list --fqbn arduino:avr:mega:cpu=atmega2560
Port Protocol Type              Board Name                FQBN                            Core       
COM3 serial   Serial Port (USB) Arduino Mega or Mega 2560 arduino:avr:mega:cpu=atmega2560 arduino:avr
```

Reply:
Oh I see, so board options can be provided without the specific flag too. So yes, it doesn't make any sense to remove it 👍

Reply:
>What's done in other cases?

The only other cases where `--fqbn` is used as a filter like it is here is the `lib examples` and `lib list` commands. In those commands, the FQBN is only used to get the platform name (e.g., `arduino:avr`) in order to find the applicable [platform bundled libraries](https://arduino.github.io/arduino-cli/dev/platform-specification/#platform-bundled-libraries) and the architecture (e.g., `avr`) to check against the [library metadata](https://arduino.github.io/arduino-cli/dev/library-specification/#library-metadata) for compatibility. So board options are actually irrelevant for those commands and they have no need of a `--board-options` flag.

So there isn't any precedent for how the match should work.

Other commands that have an `--fqbn`/`--board-options` flag (`board details`, `burn-bootloader`, `monitor`) use the default board options values (the option value defined first in the `boards.txt` file) if the user doesn't specify the option value. For example, [the `arduino:avr` platform sets the default value of the `arduino:avr:mega` board's `cpu` option to `atmega2560`](https://github.com/arduino/ArduinoCore-avr/blob/1.8.6/boards.txt#L304-L332). So `--fqbn arduino:avr:mega` is equivalent to `--fqbn arduino:avr:mega:cpu=atmega2560` or `--fqbn arduino:avr:mega --board-options "cpu=atmega2560"`